File: app-laravel/src/Shared/Application/Bus/Query.php
<?php

declare(strict_types=1);

namespace Src\Shared\Application\Bus;

/**
 * QUERY (CQRS Pattern)
 * ====================
 * Query — sistemdən data oxumaq üçün göndərilən sorğudur.
 *
 * ƏSAS QAYDALAR:
 * 1. Query heç vaxt datanı dəyişmir (side-effect free).
 * 2. Query həmişə data qaytarır.
 * 3. Query adı sual formasındadır: GetOrder, ListProducts, FindUserByEmail
 *
 * Command vs Query:
 * - CreateOrderCommand → datanı DƏYİŞİR, heç nə qaytarmır
 * - GetOrderQuery → datanı OXUYUR, Order qaytarır
 */
interface Query
{
}

File: app-spring/src/main/java/az/ecommerce/shared/application/bus/Query.java
package az.ecommerce.shared.application.bus;

/**
 * Laravel: src/Shared/Application/Bus/Query.php (marker)
 * CQRS-də read əməliyyatları (side-effect-siz, idempotent).
 *
 * NÜMUNƏ:
 *   public record GetOrderQuery(UUID orderId) implements Query<OrderDto> {}
 */
public interface Query<R> {
}
